modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/portlet/action/GetObjectDefinitionInfoMVCResourceCommand.java

```diff
@@ -57,6 +59,21 @@ protected void doServeResource(
 		JSONPortletResponseUtil.writeJSON(
 			resourceRequest, resourceResponse,
 			JSONUtil.put(
+				"isWorkflowSupported",
+				() -> {
+					WorkflowHandler<?> workflowHandler =
+						WorkflowHandlerRegistryUtil.getWorkflowHandler(
+							objectDefinition.getClassName());
+
+					if ((workflowHandler != null) &&
+						workflowHandler.isVisible()) {
+
+						return true;
+					}
+
+					return false;
+				}
+			).put(
 				"tableName", objectDefinition.getDBTableName()
 			).put(
 				"workflowDefinitionTitle",
```

Comment:
@vngoliveira please change the `workflowDefinitionTitle` to return `null` if  the workflow is not supported:

```java
boolean workflowSupported = true;

WorkflowHandler<?> workflowHandler =
	WorkflowHandlerRegistryUtil.
		getWorkflowHandler(objectDefinition.getClassName());

if ((workflowHandler == null) || !workflowHandler.isVisible()) {
	workflowSupported = false;
}

boolean finalWorkflowSupported = workflowSupported;

JSONPortletResponseUtil.writeJSON(
	resourceRequest, resourceResponse,
	JSONUtil.put(
		"isWorkflowSupported", finalWorkflowSupported
	).put(
		"tableName", objectDefinition.getDBTableName()
	).put(
		"workflowDefinitionTitle",
		() -> {
			if (!finalWorkflowSupported) {
				return null;
			}
```

Reply:
remember to update the test after this change